src/dynamicsSimulation.cpp

```diff
@@ -1357,22 +1364,47 @@ bool DynamicsSimulation::updateWalkerPositionAndHandleBouncing(Vector3d &bounced
 
         walker.status = Walker::bouncing;
 
-        tmax -= displ;
-
-        // Labels the walker w/r it's orientation.
-        if(colision.col_location == Collision::inside){
-            walker.intra_extra_consensus--;
-            walker.location = Walker::intra;
-        }
-        if(colision.col_location == Collision::outside){
-            walker.intra_extra_consensus++;
-            walker.location = Walker::extra;
-        }
-        if(walker.initial_location == Walker::unknown){
-            walker.initial_location = walker.location;
+        if(colision.col_percolation){
+          // Labels the walker w/r it's orientation.
+          if(colision.col_location == Collision::inside){
+              walker.intra_extra_consensus++;
+              walker.location = Walker::extra;
+              bounced_step=step;
+              double step_lenght_after_crossing= sqrt(6.0*params.diffusivity_ex*params.sim_duration/params.num_steps);
+              tmax= tmax + (step_lenght_after_crossing- step_lenght);
+              step_lenght= step_lenght_after_crossing;
+          }
+          if(colision.col_location == Collision::outside){
+              walker.intra_extra_consensus--;
+              walker.location = Walker::intra;
+              bounced_step=step;
+              double step_lenght_after_crossing= sqrt(6.0*params.diffusivity_ex*params.sim_duration/params.num_steps);
+              tmax= tmax + (step_lenght_after_crossing- step_lenght);
+              step_lenght= step_lenght_after_crossing;
+          }
+          if(walker.initial_location == Walker::unknown){
```

Comment:
What happens if this case happens now that you explicitly need a diffusivity intra or extra (no CSF) or other compartments